flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/RestartBackoffTimeStrategyFactoryLoader.java

```diff
@@ -116,7 +116,9 @@ private static Optional<RestartBackoffTimeStrategy.Factory> getJobRestartStrateg
                             exponentialDelayConfig.getMaxBackoff().toMilliseconds(),
                             exponentialDelayConfig.getBackoffMultiplier(),
                             exponentialDelayConfig.getResetBackoffThreshold().toMilliseconds(),
-                            exponentialDelayConfig.getJitterFactor()));
+                            exponentialDelayConfig.getJitterFactor(),
+                            RestartStrategyOptions.RESTART_STRATEGY_EXPONENTIAL_DELAY_ATTEMPTS
```

Comment:
I didn't do it because `ExponentialDelayRestartStrategyConfiguration` is deprecated since 1.19.  The `RestartStrategies` has been deprecated, and `ExponentialDelayRestartStrategyConfiguration` is the inner class of it. 

So the new option won't be supportted in the deprecated api.